rust/src/bs_safe.rs

```diff
@@ -101,9 +103,17 @@ pub enum BridgeStanError {
     /// Setting a print-callback failed.
     #[error("Failed to set a print-callback: {0}")]
     SetCallbackFailed(String),
+    #[cfg(feature = "compile-stan-model")]
```

Comment:
This is a question because I'm still relatively new to Rust: is putting specific parts of an enum behind a config a fine thing to do?

I guess the enum is already `non_exhaustive`, so it seems fine, but wanted to ask

Reply:
tricky question, according to my reading of this discussion https://www.reddit.com/r/rust/comments/xyik51/is_feature_gating_some_varaints_of_an_enum_a_bad/ it is acceptable if the enum is non_exhaustive, it's a controversial topic like, e.g. some people are against the use of box leak no matter what.

Reply:
@aseyboldt could you share your opinion on this?

Reply:
I'm not sure either. I guess I'd rather not gate it, and have the variant in the enum all the time.